sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLListener.scala

```diff
@@ -101,6 +101,8 @@ class SQLListener(conf: SparkConf) extends SparkListener with Logging {
 
   private val retainedExecutions = conf.getInt("spark.sql.ui.retainedExecutions", 1000)
 
+  private val retainedStages = conf.getInt("spark.ui.retainedStages", 1000)
```

Comment:
@tashoyan . Could you add a doc for this like `spark.sql.ui.retainedExecutions` here?
Please refer #9052.

Reply:
BTW, the name should be `spark.sql.ui.retainedStages` instead of `spark.ui.retainedStages`.

Reply:
@dongjoon-hyun , It is already documented in the same file configuration.md:
```
How many stages the Spark UI and status APIs remember before garbage collecting.
This is a target maximum, and fewer elements may be retained in some circumstances.
```
I did not involve a new parameter, I just used an existing one.
Regarding renaming to `spark.sql.ui.retainedStages`, I believe it should be done in a separate pull request - if should. This parameter is also used in other parts of Spark code, not only SQL.

Reply:
Ah. My bad. Thanks.

Reply:
@tashoyan . Since you are not introducing a new one, could you use the existing default value?
```scala
-  private val retainedStages = conf.getInt("spark.ui.retainedStages", 1000)
+  private val retainedStages =
+    conf.getInt("spark.ui.retainedStages", SparkUI.DEFAULT_RETAINED_STAGES)
```

Reply:
Done for branch-2.2